pkg/kubecost/allocation_test.go

```diff
@@ -2068,8 +2068,254 @@ func TestAllocationSetRange_Accumulate(t *testing.T) {
+		t.Fatalf("accumulating AllocationSetRange: expected first allocationSet window duration to be %v; actual %v", dur, result.allocations[0].Resolution())
+	}
+	if result.allocations[1].Resolution() != dur/2 {
+		t.Fatalf("accumulating AllocationSetRange: expected second allocationSet window duration to be %v; actual %v", dur, result.allocations[1].Resolution())
+	}
+
+	// test 4 1d sets
+	sumCost = 0.0
+	result, err = NewAllocationSetRange(ago3dAS, ago2dAS, yesterdayAS, todayAS).AccumulateBy(dur)
+	if err != nil {
+		t.Fatalf("unexpected error accumulating nil AllocationSetRange: %s", err)
+	}
+	if result == nil {
+		t.Fatalf("accumulating AllocationSetRange: expected AllocationSet; actual %s", result)
+	}
+
+	for _, as := range result.allocations {
+		sumCost += as.TotalCost()
+	}
+	if sumCost != 24.0 {
+		t.Fatalf("accumulating AllocationSetRange: expected total cost 24.0; actual %f", sumCost)
+	}
+	if result.allocations[0].Resolution() != dur {
+		t.Fatalf("accumulating AllocationSetRange: expected first allocationSet window duration to be %v; actual %v", dur, result.allocations[0].Resolution())
+	}
+	if result.allocations[1].Resolution() != dur {
+		t.Fatalf("accumulating AllocationSetRange: expected first allocationSet window duration to be %v; actual %v", dur, result.allocations[1].Resolution())
+	}
+
+	// // Accumulate non-nil with nil should result in copy of non-nil, regardless of order
+	// accumulateBy -> accumulate -> returns todayAs to set as result but nil pointer dereference error?
+	// result, err = NewAllocationSetRange(nil, todayAS).AccumulateBy(dur)
+	// if err != nil {
+	// 	t.Fatalf("unexpected error accumulating AllocationSetRange of length 1: %s", err)
+	// }
+	// if result == nil {
+	// 	t.Fatalf("accumulating AllocationSetRange: expected AllocationSet; actual %s", result)
+	// }
+
+	// for _, as := range result.allocations {
+	// 	if as.TotalCost() != 6.0 {
+	// 		t.Fatalf("accumulating AllocationSetRange: expected total cost 6.0; actual %f", as.TotalCost())
+	// 	}
+	// }
+
+	// result, err = NewAllocationSetRange(todayAS, nil).Accumulate()
+	// if err != nil {
+	// 	t.Fatalf("unexpected error accumulating AllocationSetRange of length 1: %s", err)
+	// }
+	// if result == nil {
+	// 	t.Fatalf("accumulating AllocationSetRange: expected AllocationSet; actual %s", result)
+	// }
+	// if result.TotalCost() != 6.0 {
+	// 	t.Fatalf("accumulating AllocationSetRange: expected total cost 6.0; actual %f", result.TotalCost())
+	// }
+
+	// result, err = NewAllocationSetRange(nil, todayAS, nil).Accumulate()
+	// if err != nil {
+	// 	t.Fatalf("unexpected error accumulating AllocationSetRange of length 1: %s", err)
+	// }
+	// if result == nil {
+	// 	t.Fatalf("accumulating AllocationSetRange: expected AllocationSet; actual %s", result)
+	// }
+	// if result.TotalCost() != 6.0 {
+	// 	t.Fatalf("accumulating AllocationSetRange: expected total cost 6.0; actual %f", result.TotalCost())
+	// }
```

Comment:
Are these test cases you want to keep?

Reply:
the tests were being currently worked on last push, these will be kept and are currently working. The last step of wrapping up the tests will be rewriting them in a similar format to the window expand tests